src/panels/lovelace/components/hui-card-options.ts

```diff
@@ -48,11 +48,9 @@ export class HuiCardOptions extends LitElement {
     `;
   }
   private _editCard() {
-    fireEvent(this, "show-edit-card", {
-      hass: this.hass,
-      cardConfig: this.cardConfig,
-      reloadLovelace: () => fireEvent(this, "config-refresh"),
-    });
+    showEditCardDialog(this, this.hass!, this.cardConfig!, () =>
```

Comment:
I think that this should take 2 arguments; `this` and an object with type `EditCardDialogParams`. It allows type checking and allows us to just store 1 object in the dialog class too. [Example](https://github.com/home-assistant/home-assistant-polymer/pull/2102/files#diff-014800f59089ad0b693d9f970c1f5793R30)

Reply:
Oh, you already have this with `EditDialogParams`. Well then this function just needs an object of type `EditDialogParams` passed in .